File: 项目文件/frontend/src/api/project-events.ts
import { request } from '../utils/request';
import type {
  ProjectEvent,
  ProjectEventCreate,
  ProjectEventUpdate,
  ProjectEventListResponse,
} from '../types/project-event';
import type { UnifiedResponse } from '../types/user';

export interface ProjectEventListParams {
  project_id: string;
  page?: number;
  page_size?: number;
  event_type?: string;
}

export async function listProjectEvents(
  params: ProjectEventListParams,
): Promise<UnifiedResponse<ProjectEventListResponse>> {
  const searchParams = new URLSearchParams();
  searchParams.set('project_id', params.project_id);
  if (params.page) searchParams.set('page', String(params.page));
  if (params.page_size) searchParams.set('page_size', String(params.page_size));
  if (params.event_type) searchParams.set('event_type', params.event_type);
  const query = searchParams.toString();
  return request<ProjectEventListResponse>(`/project-events/${query ? `?${query}` : ''}`);
}

export async function getProjectEvent(id: string): Promise<UnifiedResponse<ProjectEvent>> {
  return request<ProjectEvent>(`/project-events/${id}`);
}

export async function createProjectEvent(
  data: ProjectEventCreate,
): Promise<UnifiedResponse<ProjectEvent>> {
  return request<ProjectEvent>('/project-events/', { method: 'POST', body: data });
}

export async function updateProjectEvent(
  id: string,
  data: ProjectEventUpdate,
): Promise<UnifiedResponse<ProjectEvent>> {
  return request<ProjectEvent>(`/project-events/${id}`, { method: 'PUT', body: data });
}

export async function deleteProjectEvent(id: string): Promise<UnifiedResponse<null>> {
  return request<null>(`/project-events/${id}`, { method: 'DELETE' });
}

File: 项目文件/backend/app/api/project_events.py
"""项目事件 API 路由"""

from __future__ import annotations

import uuid

from fastapi import APIRouter, Depends, Query
from sqlalchemy.ext.asyncio import AsyncSession

from app.core.database import get_db
from app.core.deps import get_current_user
from app.models.user import User
from app.schemas.common import UnifiedResponse
from app.schemas.project_event import (
    ProjectEventCreate,
    ProjectEventListResponse,
    ProjectEventResponse,
    ProjectEventUpdate,
)
from app.services.project_event import (
    create_project_event,
    delete_project_event,
    get_project_event,
    list_project_events,
    update_project_event,
)

router = APIRouter()


@router.get("/", response_model=UnifiedResponse[ProjectEventListResponse])
async def list_project_events_endpoint(
    project_id: uuid.UUID = Query(...),
    page: int = Query(1, ge=1),
    page_size: int = Query(20, ge=1, le=100),
    event_type: str | None = Query(None),
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    items, total = await list_project_events(
        db, project_id, current_user, page, page_size, event_type
    )
    return UnifiedResponse(
        data=ProjectEventListResponse(
            items=[ProjectEventResponse.model_validate(i) for i in items],
            total=total,
        )
    )


@router.post("/", response_model=UnifiedResponse[ProjectEventResponse])
async def create_project_event_endpoint(
    request: ProjectEventCreate,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    item = await create_project_event(
        db, request.project_id, current_user, request.model_dump()
    )
    return UnifiedResponse(data=ProjectEventResponse.model_validate(item))


@router.get("/{event_id}", response_model=UnifiedResponse[ProjectEventResponse])
async def get_project_event_endpoint(
    event_id: uuid.UUID,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    item = await get_project_event(db, event_id, current_user)
    return UnifiedResponse(data=ProjectEventResponse.model_validate(item))


@router.put("/{event_id}", response_model=UnifiedResponse[ProjectEventResponse])
async def update_project_event_endpoint(
    event_id: uuid.UUID,
    request: ProjectEventUpdate,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    item = await update_project_event(
        db, event_id, current_user, request.model_dump(exclude_unset=True)
    )
    return UnifiedResponse(data=ProjectEventResponse.model_validate(item))


@router.delete("/{event_id}", response_model=UnifiedResponse[None])
async def delete_project_event_endpoint(
    event_id: uuid.UUID,
    current_user: User = Depends(get_current_user),
    db: AsyncSession = Depends(get_db),
):
    await delete_project_event(db, event_id, current_user)
    return UnifiedResponse(msg="项目事件已删除")
